crates/blockchain-tree/src/externals.rs

```diff
@@ -38,8 +40,38 @@ impl<N: ProviderNodeTypes, E> TreeExternals<N, E> {
         provider_factory: ProviderFactory<N>,
         consensus: Arc<dyn Consensus>,
         executor_factory: E,
+        disable_transaction_timeout: bool,
     ) -> Self {
-        Self { provider_factory, consensus, executor_factory }
+        Self { provider_factory, consensus, executor_factory, disable_transaction_timeout }
+    }
+}
+
+impl<N, E> TreeExternals<N, E>
+where
+    N: NodeTypesWithDB + ProviderNodeTypes,
+    <N as NodeTypes>::ChainSpec: EthereumHardforks,
+{
+    pub fn get_provider(
+        &self,
+    ) -> DatabaseProvider<<<N as NodeTypesWithDB>::DB as Database>::TX, <N as NodeTypes>::ChainSpec>
+    {
+        let mut provider = self.provider_factory.provider().unwrap();
+        if self.disable_transaction_timeout {
+            provider = provider.disable_long_read_transaction_safety();
+        }
+        provider
```

Comment:
we should not have `unwrap`s

Reply:
yeah, this should be more of a draft honestly, planned to remove the unwraps, just forgot